lighthouse-core/lib/i18n/i18n.js

```diff
@@ -180,36 +188,40 @@ function _collectAllCustomElementsFromICU(icuElements, seenElementsById = new Ma
  * will apply Lighthouse custom formatting to the values based on the argumentElement
  * format style.
  *
- * @param {Array<import('intl-messageformat-parser').Element>} argumentElements
+ * @param {string} icuMessage
+ * @param {Array<ArgumentElement>} argumentElements
  * @param {Record<string, string | number>} [values]
+ * @return {Record<string, string | number>}
  */
-function _processParsedElements(argumentElements, values = {}) {
-  // Throw an error if a message's value isn't provided
-  argumentElements
-    .filter(el => el.type === 'argumentElement')
-    .forEach(el => {
-      if (el.id && (el.id in values) === false) {
-        throw new Error(`ICU Message contains a value reference ("${el.id}") that wasn't provided`);
-      }
-    });
-
-  // Round all milliseconds to the nearest 10
-  argumentElements
-    .filter(el => el.format && el.format.style === 'milliseconds')
-    // @ts-ignore - el.id is always defined when el.format is defined
-    .forEach(el => (values[el.id] = Math.round(values[el.id] / 10) * 10));
-
-  // Convert all seconds to the correct unit
-  argumentElements
-    .filter(el => el.format && el.format.style === 'seconds' && el.id === 'timeInMs')
-    // @ts-ignore - el.id is always defined when el.format is defined
-    .forEach(el => (values[el.id] = Math.round(values[el.id] / 100) / 10));
-
-  // Replace all the bytes with KB
-  argumentElements
-    .filter(el => el.format && el.format.style === 'bytes')
-    // @ts-ignore - el.id is always defined when el.format is defined
-    .forEach(el => (values[el.id] = values[el.id] / 1024));
+function _processParsedElements(icuMessage, argumentElements, values = {}) {
+  for (const {id, format} of argumentElements) {
+    // Throw an error if a message's value isn't provided
+    if (id && (id in values) === false) {
+      // eslint-disable-next-line max-len
+      throw new Error(`ICU Message "${icuMessage}" contains a value reference ("${id}") that wasn't provided`);
+    }
+
+    // Direct `{id}` replacement and non-numeric values need no formatting.
+    if (!format || format.type !== 'numberFormat') continue;
+
+    const value = values[id];
+    if (typeof value !== 'number') {
+      // eslint-disable-next-line max-len
+      throw new Error(`ICU Message "${icuMessage}" contains a numeric reference ("${id}") but provided value was not a number`);
```

Comment:
makes tsc happy to prove these are numbers (for the formatting divisions below) and it seems like a reasonable error condition to call out